src/main/java/smartthings/konsumer/circuitbreaker/SimpleCircuitBreaker.java

```diff
@@ -2,42 +2,34 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import smartthings.konsumer.event.KonsumerEvent;
-import smartthings.konsumer.event.KonsumerEventListener;
-import smartthings.konsumer.util.ThreadFactoryBuilder;
 
 import javax.annotation.concurrent.GuardedBy;
 import javax.annotation.concurrent.ThreadSafe;
 import java.util.*;
-import java.util.concurrent.*;
 
 @ThreadSafe
 public class SimpleCircuitBreaker implements CircuitBreaker {
-
 	private final static Logger log = LoggerFactory.getLogger(SimpleCircuitBreaker.class);
 
 	private final Object mutex = new Object();
 
 	@GuardedBy("mutex")
 	private boolean open = false;
-
 	@GuardedBy("mutex")
 	private final Set<String> sources = new HashSet<>();
 
-	private final List<KonsumerEventListener> listeners = new CopyOnWriteArrayList<>();
-
-	private final ExecutorService callbackExecutor;
+	private CircuitBreakerListener listener;
 
-	public SimpleCircuitBreaker() {
-		this(Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
-			.setDaemon(true)
-			.setNameFormat("KafkaCircuitBreaker-callback")
-			.build())
-		);
+	@Override
+	public void init(CircuitBreakerListener listener) {
+		synchronized (mutex) {
+			this.listener = listener;
```

Comment:
I did it this way for two reasons. I was trying to make the lifecycle callbacks consistent across the circuitbreaker and the filters. 
Also, trying to avoid registering listeners in the constructors since it could (at least on theory) allow the callbacks to be called before the object is fully constructed. Doing it this way lets KafkaListener register the listener in the run method.